src/main/scala/com/fulcrumgenomics/umi/ConsensusCallingIterator.scala

```diff
@@ -63,41 +64,43 @@ class ConsensusCallingIterator[ConsensusRead <: SimpleRead](sourceIterator: Iter
     }
     else {
       val halfMaxRecords   = maxRecordsInRam / 2
-      val groupingIterator = new SamRecordGroupedIterator(new AsyncIterator(progressIterator, Some(halfMaxRecords)).start(), caller.sourceMoleculeId)
       val pool             = new ForkJoinPool(threads - 1, ForkJoinPool.defaultForkJoinWorkerThreadFactory, null, true)
-      val bufferedIter     = groupingIterator.bufferBetter
       val callers          = new IterableThreadLocal[UmiConsensusCaller[ConsensusRead]](() => caller.emptyClone())
+      val groupingIterator = {
+        val async   = AsyncIterator(progressIterator, Some(halfMaxRecords))
```

Comment:
I run into this so frequently with our async classes, namely forgetting to start them.  Perhaps the behavior should be changed to always start, but have a way to say "don't start"
```suggestion
        val async   = AsyncIterator(progressIterator, Some(halfMaxRecords)).start()
```

Reply:
ffs, and I got it wrong too, since you changed this from the primary constructor to a secondary constructor, with the latter automatically starting!

Reply:
Yeah, for better or worse, we decided the apply() methods should start them, while the constructors shouldn't.  Perhaps we should just change them to have the constructors take a non-defaulted `start: Boolean` so it's super clear.